test/unit/ngsiv2/ngsiService/subscriptions-test.js

```diff
@@ -0,0 +1,351 @@
+                        done();
+                    });
+                });
+            });
+        });
+    });
+    describe('When a client invokes the unsubscribe() function for an entity', function() {
+        beforeEach(function(done) {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .delete('/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8')
+                .reply(204);
+
+            done();
+        });
+        it('should delete the subscription from the CB', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    iotAgentLib.unsubscribe(device, '51c0ac9ed714fb3b37d7d5a8', function(error) {
+                        iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+
+                            contextBrokerMock.done();
+                            done();
+                        });
+                    });
+                });
+            });
+        });
+        it('should remove the id from the subscriptions array', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    iotAgentLib.unsubscribe(device, '51c0ac9ed714fb3b37d7d5a8', function(error) {
+                        iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                            should.not.exist(error);
+                            should.exist(device);
+                            should.exist(device.subscriptions);
+                            device.subscriptions.length.should.equal(0);
+                            done();
+                        });
+                    });
+                });
+            });
+        });
+    });
+    describe('When a client removes a device from the registry', function() {
+        beforeEach(function(done) {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .delete('/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8')
+                .reply(204);
+
+            done();
+        });
+
+        it('should delete the subscription from the CB', function(done) {
+            iotAgentLib.getDevice('MicroLight1', 'smartGondor', '/gardens', function(error, device) {
+                iotAgentLib.subscribe(device, ['attr_name'], null, function(error) {
+                    iotAgentLib.unregister(device.id, 'smartGondor', '/gardens', function(error) {
+                        contextBrokerMock.done();
```

Comment:
Similar comment to https://github.com/dcalvoalonso/iotagent-node-lib/pull/4/files#r160198941

Reply:
Please check https://github.com/dcalvoalonso/iotagent-node-lib/blob/task/bidirectionalityPlugin/test/unit/ngsiv2/ngsiService/subscriptions-test.js#L190

Reply:
NTC